writer.go

```diff
@@ -257,6 +267,7 @@ func (p *MasterPlaylist) String() string {
 func NewMediaPlaylist(winsize uint, capacity uint) (*MediaPlaylist, error) {
 	p := new(MediaPlaylist)
 	p.ver = minver
+	p.dirty = true
```

Comment:
Does this need to be protected by the lock ?

Reply:
No, it is constructor.

Reply:
Maybe I was a little too trigger-happy when I merged the code.  Is this comment resolved?  @mk-livepeer I'm not sure if I understand the comment here - there isn't a lock statement in this block of code.  Can you explain more?

Reply:
@ericxtang As I am still new to the codebase, I raised this comment as a question.  "Does this need to be protected by the lock?"  ...  Ivan explained that this particular function is a constructor -- this function does not need to be protected by the lock - all is well.